Geometry_Engine/Query/GeometryHash.cs

```diff
@@ -0,0 +1,381 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.oM.Base;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Security.Cryptography.Xml;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("All geometry hash methods are implemented to be translational.")]
+        public static double[] IGeometryHash(this IBHoMObject bhomObj)
+        {
+            IGeometry igeom = bhomObj.IGeometry();
+
+            return GeometryHash(igeom as dynamic, 0);
+        }
+
+        private enum TypeTranslationFactor
+        {
+            Point = 0,
+            Arc,
+            Circle,
+            Ellipse,
+            Line,
+            NurbsCurve,
+            PlanarSurface,
+            Extrusion,
+            Loft,
+            Pipe,
+            PolySurface,
+            NurbsSurface,
+            SurfaceTrim,
+            Mesh,
+            Mesh3D,
+            Plane
```

Comment:
For some reason, when looking at this type list, my geometry brain sees this as the logical order - seemingly raising in complexity/degree of freeform.

In fact this is the order I would probably list the geometry if I was introducing their definitions in documentation. So makes sense to me at least! 😸 😕 

But offering this as a suggestion, happy to hear others alternate orders - it is all arbitrary in the end after all! @alelom @IsakNaslundBh 

```suggestion
            Point = 0,
            Plane,
            Line,
            Arc,
            Circle,
            Ellipse,
            NurbsCurve,
            PlanarSurface,
            Pipe,
            Extrusion,
            Loft,
            NurbsSurface,
            SurfaceTrim,
            Mesh,
            Mesh3D,
```

Reply:
Looks good to me! 😄 As I mentioned, the only reason I followed that specific order was to avoid confusion when writing down each type, following the order they appeared in in the Engine folder -- no reason not to make it more tidy now!